README.md

```diff
@@ -19,10 +20,23 @@ via autowiring.
 
 ## Features
 
-- Invoke callable or create an object of a given class.
-- Resolve dependencies by parameter types using the given [PSR-11](http://www.php-fig.org/psr/psr-11/) container.
-- Pass concrete dependency instances by type.
-- Pass arguments by name.
+Injector can automatically resolve and inject dependencies when invoking
+methods and creating objects. It uses autowiring based on [type
+declarations](https://www.php.net/manual/en/functions.arguments.php#functions.arguments.type-declaration)
+and/or argument names.
+
+Dependencies are fetched from two sources:
+
+ * A PSR-11 compatible container that provides object instances and uses
+   classnames as ids
+ * A given list of arguments that optionally includes argument names
+
+Dependencies are then resolved in this order:
```

Comment:
The order is more complex and is provided in full docs: https://github.com/yiisoft/injector/tree/master/docs/en#how-it-works

Reply:
Ok changed it again and removed details on how things are resolved.

I've also redone the flow chart as the labels were a bit confusing to me at first read. You can check it here: https://drive.google.com/file/d/1hVY79NvOWblgzAIbaIYhaBobIpAs-FZJ/view?usp=sharing. If it's OK I can add another PR to update the image in the docs. I can also provide the edit link to diagrams.net - but this should probably better not be public.

Reply:
The diagram looks alright. In fact, there are source files for it (`*.graphml`). These could be edited with [yEd](https://www.yworks.com/products/yed).

Reply:
Part of the original diagram is missing:

![image](https://user-images.githubusercontent.com/47294/95082198-5d918600-0723-11eb-9415-cc4fc4e4e124.png)

Reply:
Other than that, it is definitely cleaner.